src/components/ScenarioParameters/ScenarioParameters.js

```diff
@@ -245,12 +245,15 @@ const ScenarioParameters = ({
     updateAndLaunchScenario(workspaceId, scenarioId, parametersData);
     changeEditMode(false);
   };
-  const fileUploadComponent = UploadFileUtils.constructFileUpload('0',
+  const fileUploadComponent = UploadFileUtils.constructFileUpload(
+    '0',
     initialStockFile,
     setInitialStockFile,
     initialStockDataset.id,
     INITIAL_STOCK_PARAM.defaultFileTypeFilter,
-    editMode);
+    editMode,
+    scenarioId
```

Comment:
I'm not a huge fan of carrying the `scenarioId` information in a graphical component that does not need it, only for tracking purposes. Can we store this kind of data in the `AppInsights` service singleton instead ?

For example, we could add a `setCurrentScenarioId` to the `AppInsights` service, and call this method every time the user selects a different scenario. The scenario id value could then be used in all events and metrics that need it.